backend/src/main/scala/com/softwaremill/bootzooka/passwordreset/PasswordResetService.scala

```diff
@@ -38,18 +37,26 @@ class PasswordResetService(
   private def createCode(user: User): ConnectionIO[PasswordResetCode] = {
     logger.debug(s"Creating password reset code for user: ${user.id}")
     val validUntil = clock.instant().plus(config.codeValid.toMinutes, ChronoUnit.MINUTES)
-    val code = PasswordResetCode(idGenerator.nextId[PasswordResetCode](), user.id, validUntil)
-    passwordResetCodeModel.insert(code).map(_ => code)
+    for {
+      id <- idGenerator.nextId[PasswordResetCode]().to[ConnectionIO]
+      passwordResetCode = PasswordResetCode(id, user.id, validUntil)
+      connection <- passwordResetCodeModel.insert(passwordResetCode).map(_ => passwordResetCode)
+    } yield connection
   }
 
   private def sendCode(user: User, code: PasswordResetCode): ConnectionIO[Unit] = {
     logger.debug(s"Scheduling e-mail with reset code for user: ${user.id}")
-    emailScheduler(EmailData(user.emailLowerCased, prepareResetEmail(user, code)))
+    for {
+      mail <- prepareResetEmail(user, code).to[ConnectionIO]
+      conn <- emailScheduler(EmailData(user.emailLowerCased, mail))
+    } yield conn
   }
 
-  private def prepareResetEmail(user: User, code: PasswordResetCode): EmailSubjectContent = {
-    val resetLink = String.format(config.resetLinkPattern, code.id)
-    emailTemplates.passwordReset(user.login, resetLink)
+  private def prepareResetEmail(user: User, code: PasswordResetCode): Task[EmailSubjectContent] = {
```

Comment:
does this need to be wrapped in a `Task`?

Reply:
not anymore, thanks for point that out